src/main/java/jenkins/scm/impl/TrustworthyBuilds.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2022 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.scm.impl;
+
+import hudson.Extension;
+import hudson.model.Cause;
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.User;
+import jenkins.scm.api.TrustworthyBuild;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class TrustworthyBuilds {
+
+    // Also effectively handles ReplayCause since that is only ever added in conjunction with UserIdCause. (see ReplayAction.run2)
+    @Extension
+    public static TrustworthyBuild byUserId() {
+        return (build, listener) -> {
+            var cause = build.getCause(Cause.UserIdCause.class);
```

Comment:
While it should be rare, there can be multiple causes of the same type for a build that aren't collapsed into one entry. Whether a build is approved or not depends on the (insertion) order of the `CauseAction#causeBag`.

Reply:
> there can be multiple causes of the same type for a build that aren't collapsed into one entry

The API does not prevent it but this ought never happen—it is the responsible of code triggering the build to pass at most one `Cause` of any given type. At worst a build is not considered trusted when it _could_ have been, so this does not seem like a problem in practice.